sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -98,13 +98,15 @@ object FakeV2SessionCatalog extends TableCatalog {
  *                            views.
  * @param nestedViewDepth The nested depth in the view resolution, this enables us to limit the
  *                        depth of nested views.
+ * @param maxNestedViewDepth The maximum allowed depth of nested view resolution.
  * @param relationCache A mapping from qualified table names to resolved relations. This can ensure
  *                      that the table is resolved only once if a table is used multiple times
  *                      in a query.
  */
 case class AnalysisContext(
     catalogAndNamespace: Seq[String] = Nil,
     nestedViewDepth: Int = 0,
+    maxNestedViewDepth: Int = SQLConf.get.maxNestedViewDepth,
```

Comment:
Is `maxNestedViewDepth` sticks to the value of the first created AnalysisContext in the current thread? Shall we make `AnalysisContext.value` without an initialValue?

Reply:
The initial value is created with `AnalysisContext()`, so `maxNestedViewDepth` starts with the configured value. For nested view resolution, we create new contexts with `withAnalysisContext`, which retain the value of `maxNestedViewDepth`